### PR TITLE
refactor(cat-voices): user and wallet model

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

- Introduces concept of `Profile` and `User` having multiple profiles. At the moment only one can be passed.
- `WalletHeader` and `WalletMeta` classes to represent information's about wallet that does not come from wallet extension.
- deletes `DummyCardanoWallet`

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Comment by damian-molinski** [comment]
This is just proposal. We'll need a way to represent wallets that won't come from extension

File: catalyst_voices/packages/catalyst_voices_models/lib/src/user/profile.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import 'package:equatable/equatable.dart';
+
+/// Defines singular profile used by [User]. One [User] may have multiple
+/// [Profile]'s.
+final class Profile extends Equatable {
+  const Profile({
+    required this.walletHeader,
+  });
+
+  final WalletHeader walletHeader;
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/lib/src/user/profile.dart:12]: fields above constructors please

File: catalyst_voices/packages/catalyst_voices_models/lib/src/user/user.dart [hunk 2]
```diff
@@ -1,17 +1,26 @@
+import 'package:catalyst_voices_models/src/user/profile.dart';
 import 'package:equatable/equatable.dart';
 
-/// Defines the profile of the app user.
+/// Defines user or the app.
 final class User extends Equatable {
-  final String name;
+  final List<Profile> profiles;
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/lib/src/user/user.dart:6]: not entirely sure why a user would have multiple profiles... they can belong to multiple teams (in the future, etc) but what are multiple profiles?

File: catalyst_voices/packages/catalyst_voices_models/lib/src/wallet/wallet_header.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import 'package:catalyst_voices_models/src/wallet/wallet_metadata.dart';
+import 'package:equatable/equatable.dart';
+
+/// This class represents basic information's about Wallet,
+/// such as [CardanoWallet], but without specifying way to take action like
+/// enable.
+///
+/// Instance of this class can come from different places like backend
+/// or user interacting with wallet extension.
+final class WalletHeader extends Equatable {
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/lib/src/wallet/wallet_header.dart:12]: Header seems UI-ish, models should be clean of UI. What about calling it a WalletInfo, etc?

Comment (damian-molinski) [reply catalyst_voices/packages/catalyst_voices_models/lib/src/wallet/wallet_header.dart:12]: I was thinking more like http headers but okey will think about smth else :)

Comment (damian-molinski) [reply catalyst_voices/packages/catalyst_voices_models/lib/src/wallet/wallet_header.dart]: Changed to `WalletInfo`

File: catalyst_voices/packages/catalyst_voices_models/lib/src/user/user.dart (hunk 2 above)
Comment (dtscalac) [reply catalyst_voices/packages/catalyst_voices_models/lib/src/user/user.dart:6]: Okay I see your point, though it's not defined if users will create separate keychains per brand or reuse the same one.

File: catalyst_voices/packages/catalyst_voices_blocs/lib/src/registration/cubits/wallet_link_cubit.dart [hunk 4]
```diff
@@ -14,7 +15,7 @@ final _logger = Logger('WalletLinkCubit');
 abstract interface class WalletLinkManager {
   Future<void> refreshWallets();
 
-  Future<bool> selectWallet(CardanoWallet wallet);
+  Future<bool> selectWallet(WalletMetadata mete);
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_blocs/lib/src/registration/cubits/wallet_link_cubit.dart:18]:
```suggestion
  Future<bool> selectWallet(WalletMetadata meta);
```

**Review by dtscalac — APPROVED** [review]
LGTM

**Pull request merged**